### PR TITLE
Keyboard/bear65 v2

### PR DESCRIPTION
**Pull request by beelzebubi**
Added support for the Bear65 V2 board by Jacky Studio + custom user script and keymap

## Description

The V2 of the Bear65 comes with a WK version, while the V1 was a WKL only board. 

This PR implements a rev2 subfolder, adding support for all version of the V2 board while leaving the original files in a rev1 folder.

Additionally, this PR adds a custom keymap for the V2 including user scripts.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
userspace additions should not happen in keyboard PRs. The changes should be removed and submitted separately. We are a little less strict on user keymaps, however if the `nazu` keymap depends on userspace, then that should also be removed.

File: keyboards/jacky_studio/bear_65/rev1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev1/rules.mk:6]:
```suggestion
```

File: keyboards/jacky_studio/bear_65/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev2/rules.mk:6]:
```suggestion
```

File: keyboards/jacky_studio/bear_65/rev2/info.json [hunk 3]
```diff
@@ -0,0 +1,322 @@
+{
+    "keyboard_name": "Bear65 V2",
+    "manufacturer": "Jacky",
+    "url": "https://qmk.fm/keyboards",
+    "maintainer": "qmk",
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev2/info.json:5]:
```suggestion
    "maintainer": "qmk",
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/jacky_studio/bear_65/rev1/info.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
+{
+    "keyboard_name": "Bear 65",
+    "manufacturer": "Jacky",
+    "url": "https://qmk.fm/keyboards",
+    "maintainer": "qmk",
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev1/info.json:5]:
```suggestion
    "maintainer": "qmk",
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/jacky_studio/bear_65/rev1/rev1.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	// LED Index to Physical Position
+	{224, 32}, {224, 10}, {192, 10}, {128, 25},
+    {70, 10}, {16, 10}, {16, 32}, {16, 64},
+    {64, 64}, {112, 64}, {176, 64}, {224, 64}
+}, {
+	// LED Index to Flag
+	LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL,
+	LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL
+} };
+#endif
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev1/rev1.c:37]: This code is no different from what's in `bear65.c`? It shouldn't be added at the revision level unless necessary.

File: keyboards/jacky_studio/bear_65/rev2/rev2.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	// LED Index to Physical Position
+	{224, 32}, {224, 10}, {192, 10}, {128, 25},
+    {70, 10}, {16, 10}, {16, 32}, {16, 64},
+    {64, 64}, {112, 64}, {176, 64}, {224, 64}
+}, {
+	// LED Index to Flag
+	LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL,
+	LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL, LED_FLAG_ALL
+} };
+#endif
```
Comment (waffle87) [inline keyboards/jacky_studio/bear_65/rev2/rev2.c:37]: Above comment.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by beelzebubi** [comment]
Do I still need to make any updates on the code?

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**